lib/cimp/kv.go

```diff
@@ -0,0 +1,117 @@
+package cimp
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/go-yaml/yaml"
+	"github.com/hashicorp/consul/api"
+)
+
+type KV map[string]interface{}
+
+func NewKV() KV {
+	kv := KV(make(map[string]interface{}))
+	return kv
+}
+
+func (kv KV) FillFromFile(path string, format FileFormat) error {
+	fileData, err := ioutil.ReadFile(path)
+	if err != nil {
+		return fmt.Errorf("read file data: %v", err)
+	}
+
+	rawData, err := UnmarshalWithFormat(format, fileData)
+	if err != nil {
+		return fmt.Errorf("unmarshal %q-file: %w", format, err)
+	}
+
+	return kv.fillRecursive("", rawData)
+}
+
+func (kv KV) Fill(prefix string, rawData map[interface{}]interface{}) error {
+	return kv.fillRecursive(prefix, rawData)
+}
+
+func (kv KV) Check(key string) bool {
+	_, ok := kv[key]
+	return ok
+}
+
+func (kv KV) GetString(key string) (string, error) {
+	value, ok := kv[key]
+	if !ok {
+		return "", fmt.Errorf("value %q: %w", key, ErrorNotFoundInKV)
+	}
+
+	switch typed := value.(type) {
+	case string:
+		return typed, nil
+	}
+
+	return "", fmt.Errorf("value %q: %w", key, ErrorTypeIncorrect)
+}
+
+func (kv KV) AddPair(pair api.KVPair) {
+	kv[pair.Key] = pair.Value
+}
+
+func (kv KV) AddPrefix(prefix string) {
+	newKV := NewKV()
```

Comment:
it's safe to delete a visited element from a loop, so we don't have to create one more map instance

Reply:
never mind :)

Reply:
we cant insert in a loop

Reply:
yup, we can get keys "a/b" and "b". With prefix "a/" it'll be a problem